Time-Dependent-Coupled-Cluster/td-ccsd.py

```diff
@@ -0,0 +1,798 @@
+# A simple Psi4 script to compute TD-CCSD from a CCSD reference
+# Scipy and numpy python modules are required
+# This script runs a post-CCSD computation, using the ERI
+# tensor, the T1 and T2 amplitudes as well as the F and W 
+# intermediates.
+#
+# CCSD
+# Created by: Daniel G. A. Smith
+# Date: 7/29/14
+# License: GPL v3.0
+#
+# TD-CCSD
+# Created by: Daniel R. Nascimento
+# Date: 4/22/17
+# License: GPL v3.0
+#
+
+import time
+import numpy as np
+np.set_printoptions(precision=14, linewidth=200, suppress=True)
+import psi4
+
+# Set memory
+psi4.set_memory(int(2e9), True)
+psi4.core.set_output_file('output.dat', True)
+
+numpy_memory = 2
+
+mol = psi4.geometry("""
+#O
+#H 1 1.1
+#H 1 1.1 2 104
+#H 0.0 0.0 0.0
+#H 1.0 0.0 0.0
+#O          0.000000000000     0.000000000000    -0.068516219310
+#H          0.000000000000    -0.790689573744     0.543701060724
+#H          0.000000000000     0.790689573744     0.543701060724
+C	0.0	0.0	0.0
+O	0.0	0.0	1.0
+symmetry c1
+no_reorient
+""")
+
+psi4.set_options({'basis': 'sto-3g',
+                  'scf_type': 'pk',
+                  'mp2_type': 'conv',
+                  'freeze_core': 'false',
+                  'e_convergence': 1e-14,
+                  'd_convergence': 1e-14})
+
+# CCSD Settings
+E_conv = 1.e-14
+maxiter = 50
+print_amps = False
+compare_psi4 = False
+
+# TD-CCSD Settings
+### defines the field polarization (0 == x, 1 == y, 2 == z)
+polarization = 1
+### sets the total number of time steps
+steps_total = 20000
+### sets the length of the time step
+time_step = 0.05
+
+# First compute RHF energy using Psi4
+scf_e, wfn = psi4.energy('SCF', return_wfn=True)
+
+# Grab data from
+C = wfn.Ca()
+ndocc = wfn.doccpi()[0]
+nmo = wfn.nmo()
+SCF_E = wfn.energy()
+eps = np.asarray(wfn.epsilon_a())
+
+# Compute size of SO-ERI tensor in GB
+ERI_Size = (nmo ** 4) * 128e-9
+print('\nSize of the SO ERI tensor will be %4.2f GB.' % ERI_Size)
+memory_footprint = ERI_Size * 5.2
+if memory_footprint > numpy_memory:
+    clean()
+    raise Exception("Estimated memory utilization (%4.2f GB) exceeds numpy_memory \
+                    limit of %4.2f GB." % (memory_footprint, numpy_memory))
+
+# Integral generation from Psi4's MintsHelper
+t = time.time()
+mints = psi4.core.MintsHelper(wfn.basisset())
+H = np.asarray(mints.ao_kinetic()) + np.asarray(mints.ao_potential())
+
+print('\nTotal time taken for ERI integrals: %.3f seconds.\n' % (time.time() - t))
+
+#Make spin-orbital MO
+print('Starting AO -> spin-orbital MO transformation...')
+t = time.time()
+MO = np.asarray(mints.mo_spin_eri(C, C))
+
+# Update nocc and nvirt
+nso = nmo * 2
+nocc = ndocc * 2
+nvirt = nso - nocc
+
+# Make slices
+o = slice(0, nocc)
+v = slice(nocc, MO.shape[0])
+
+#Extend eigenvalues
+eps = np.repeat(eps, 2)
+Eocc = eps[o]
+Evirt = eps[v]
+
+print('..finished transformation in %.3f seconds.\n' % (time.time() - t))
+
+# DPD approach to CCSD equations
+# See: http://sirius.chem.vt.edu/wiki/doku.php?id=crawdad:programming
```

Comment:
Might be worth adding a "References" section to the header of this script, with links to [the CCSD project](http://sirius.chem.vt.edu/wiki/doku.php?id=crawdad:programming:project5) and [the Stanton paper](http://sirius.chem.vt.edu/wiki/lib/exe/fetch.php?media=crawdad:programming:jcp_94_4334_1991.pdf) that these terms are defined in.  Or, in the README for the directory.